lib/ukfile/include/uk/pollqueue.h

```diff
@@ -0,0 +1,584 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+/* Multi-event poll/wait queue with update chaining support */
+
+#ifndef __UKFILE_POLLQUEUE_H__
+#define __UKFILE_POLLQUEUE_H__ 1
```

Comment:
`#include <uk/file/pollqueue.h>` makes most sense IMO.
Moving pollqueue into its own lib poses some subtle Kconfig issues, but more importantly would only be valuable if it can be used independently from files. At the moment I don't see such an application.
Moving under `<uk/file/...>` makes sense for direct consumers outside of `ukfile` (mostly epoll) because it expresses intent more clearly: "Epoll wants to directly interface with the API of a file's pollqueue", and not just "of any pollqueue".